client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -826,6 +855,14 @@ protected final ElasticsearchStatusException parseResponseException(ResponseExce
         }
     }
 
+    private <Resp> Resp processResponse(Resp resp, int statusCode) {
```

Comment:
I see the problem here: the response doesn't print out the exists flag, which should be assigned based on the status code. I would not check instanceof though here, I think it's possible to provide a context to the fromXContent method as an additional argument (currently declared Void so it is always null), make that a Boolean . It is not very easy though to make that work based on the status in the client. As an alternative, we could change the output of the API and make it print out the flag as well, like many other API do. @nik9000 what do you think?

Reply:
Could you make the `responseConverter` something like
```
response -> {
  ExplianResponse r = parseEntity(response.getEntity(), ExplainResponse::fromXContent);
  r.setExists(convertExistsResponse(response));
  return r;
}
```
?

Reply:
Ah! I see, you want the boolean as the context so you don't need to add `setExists`. You could do that like this:
```
response -> {
  CheckedFunction<XContentParser, Resp, IOException> entityParser = parser -> ExplainResponse.fromXContent(convertExistsResponse(response), parser);
  return parseEntity(response.getEntity(), entityParser);
}
```

You make ExplainResponse.fromXContent like this instead of how you have it.
```
    public static ExplainResponse fromXContent(boolean exists, XContentParser parser) {
        return PARSER.apply(parser, exists);
    }
```

I think that isn't *too* bad.

Reply:
And you'd have to change the parser declaration to:
```
private static final ConstructingObjectParser<ExplainResponse, Boolean> PARSER = new ConstructingObjectParser<>("explain", true, 
        (args, exists) -> new ExplainResponse((String) args[0], (String) args[1], (String) args[2], exists, (Explanation) args[3],
+            (GetResult) arg[4]));
```

Or something like that. I'm not 100% sure on that bit to be honest, but *somewhere* in the building function you'd use `exists`.
And then you can